cli/hosting.md

```diff
@@ -0,0 +1,34 @@
+
+# Overview
+The `amplify publish`command is designed to build and publish both the backend and the frontend of the project. Depending on the stage that the project is at, the command can be configured to publish either to a Dev or a Prod environment.<br/>
+In the current implementation, the frontend publish is only available for javascript project for static web hosting. This is acomplished by the category plugin amplify-category-hosting, using Amazon S3 and CloudFront. <br/>
+The amplify-category-hosting module uses the amplify-provider-awscloudformation to create and update the S3 and CloudFront resources.
+For more  information of the Amazon S3 and CloudFront, check their docs:<br/>
+[S3 static web hosting](https://docs.aws.amazon.com/AmazonS3/latest/dev/WebsiteHosting.html)<br/>
+[CloudFront Dev Guide](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Introduction.html)
+***
+# Workflow
+- `amplify hosting add`<br/>
+This adds the hosting resources to the backend. The command will first prompt for environment selection, either Dev or Prod. Upon completion, the cloudformation template for the resources is placed in the amplify/backend/hosting directory. <br/><br/>
+- `amplify hosting configure`<br/>
+This command walks through the steps to configure the different sections of the resources used in hosting, including S3, CloudFront, and publish ignore. See below for more details.<br/><br/>
+- `amplify publish`<br/>
+This command first builds and pushes the update of backend resources to the cloud (including the resources used in hosting), and then builds and publishes the frontend.<br/>
+For the amplify-category-hosting implementation, the frontend build artifacts will be uploaded to the S3 hosting bucket, and then if the CloudFront is used and the command is executed with the `--invalidateCloudFront` or `-c` flag, a invalidation request will be sent to the CloudFront to invalidate its cache. 
+
+# Dev and Prod
+For the amplify-category-hosting implementation, this is the default
+- Dev: only S3 static web hosting
+- Prod: S3 and CloudFront
+
+The CloudFront can be added or removed by the `amplify hosting configure` command. <br/>
+It take a long time to create and update the CloudFront in the cloud, easily 15 minutes and above, so it is generally not recommended to add CloudFront during the Dev stage. 
```

Comment:
Rewording:

CloudFront can be added or removed in your project at any time by the `amplify hosting configure` command. 
It can take time to create and replicate a CloudFront Distribution across the global CDN footprint, in some cases 15 minutes or more. Therefore the Amplify CLI provides a DEV configuration with an S3 static site only only when prototyping your application and a PROD configuration when you are ready to deploy in production. Note that the DEV stage using S3 static sites does not have full HTTPS end to end so it is only recommended for prototyping your app.